core/src/test/scala/com/acervera/osm4scala/DenseNodesIteratorSpec.scala

```diff
@@ -28,42 +28,72 @@ package com.acervera.osm4scala
 import com.acervera.osm4scala.model.{Info, NodeEntity}
 import org.openstreetmap.osmosis.osmbinary.osmformat.{DenseNodes, StringTable}
 import org.scalatest.matchers.should.Matchers
+import org.scalatest.prop.TableDrivenPropertyChecks
 import org.scalatest.wordspec.AnyWordSpec
 
 import java.io.FileInputStream
 import java.time.Instant
 
-class DenseNodesIteratorSpec extends AnyWordSpec with Matchers {
+class DenseNodesIteratorSpec extends AnyWordSpec with Matchers with TableDrivenPropertyChecks {
+
+  private val strTable = StringTable parseFrom new FileInputStream(
+    "core/src/test/resources/com/acervera/osm4scala/primitives/dense/strTable"
+  )
+  private val osmosisDense = DenseNodes parseFrom new FileInputStream(
+    "core/src/test/resources/com/acervera/osm4scala/primitives/dense/dense"
+  )
+
+  private val osmosisDenseVisibleTrue = osmosisDense.copy(denseinfo =
+    Option(osmosisDense.denseinfo.get.copy(visible = Vector.fill(osmosisDense.id.size)(true)))
+  )
+
+  private val expectedDenseNodes =
+    Table(
+      ("testName", "osmosisDense", "visible"),
+      ("with visible flag unset", osmosisDense, None),
+      ("with visible flag explicitly set to true", osmosisDenseVisibleTrue, Some(true))
+    )
 
   "The DenseNodesIterator should" should {
-    "Read known node" in {
+    forAll(expectedDenseNodes) { (testName, osmosisDense, visible) =>
+      s"Read known node $testName" in {
+        DenseNodesIterator(strTable, osmosisDense)
+          .find(_.id == 4020124946L) shouldBe Some(
+          NodeEntity(
+            id = 4020124946L,
+            latitude = 43.732560499999984,
+            longitude = 7.418018399999998,
+            tags = Map("entrance" -> "yes", "addr:street" -> "Rue de la Colle", "addr:housenumber" -> "4"),
+            info = Some(
+              Info(
+                version = Some(1),
+                timestamp = Some(Instant.parse("2016-02-22T17:20:29Z")),
+                changeset = Some(0),
+                userId = Some(0),
+                userName = Some(""),
+                visible = visible
+              )
+            )
+          )
+        )
+      }
+    }
+  }
```

Comment:
One small comment. I can see the `"The DenseNodesIterator should" should` case description repeated. Maybe will be better if you remove lines 80 to 82 (both inclusive) to group both in the same `should`. What do you think?

Reply:
Hi, I intentionally left them as 2 separate test cases because the first test is parametrized and tests the attributes of a `NodeEntity` and the second test checks if an exception is thrown.

Because the first test is parametrized it seemed logical to me to put the exception checking in a different test. But I have no problem putting it in 1 test though! I'll follow your opinion

Reply:
I get your point. What I try to avoid is to have repeated the same description because then, the description is not useful.

What about something like:
```scala
"The DenseNodesIterator" should {
   "Read known node" when
      forAll(expectedDenseNodes) {
         (testName, osmosisDense, visible) => testName in {}
    }
   "Throw an exception when non-visible nodes are included" in {}
}
```

BTW: Could you remove the "should" text in the description "The DenseNodesIterator **should**"? I had it, but it is wrong (It will appear two times when running). 

Anyway, I don't want to bother you too much. I appreciate your help.

Reply:
Ah yes, I did not know it was possible to group them in 1 test like that. I put them together in one test and removed the `should`